pkg/controller/storagecluster/prometheus.go

```diff
@@ -0,0 +1,112 @@
+package storagecluster
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sYAML "k8s.io/apimachinery/pkg/util/yaml"
+)
+
+const (
+	internalPrometheusRuleFilepath = "/ocs-prometheus-rules/prometheus-ocs-rules.yaml"
+	externalPrometheusRuleFilepath = "/ocs-prometheus-rules/prometheus-ocs-rules-external.yaml"
```

Comment:
Why are you storing them as YAML files instead of building them as go structs? Especially given the minimal differences between the two?

Reply:
> Why are you storing them as YAML files instead of building them as go structs?
>
That's what every monitoring project does.
It's much easier to store a YAML and apply it than to create Go structs for each alert.
Also, YAML files are more flexible in terms of updating as required and adding more alerts. With Go struct, you'll have to build binary each time there is even a small change in alert. Decoupling is very helpful and scalable.

> Especially given the minimal differences between the two?
>
That minimal difference between the two is what makes them work on Internal and External cluster.

You might not see it now as this is just the tip of the iceberg but soon you'll see why YAML (separate for Internal and External mode) is a better option.  Specially, once we add more and more alerts and the automation to generate these YAML files.

Also, if we store YAML files, we can just `oc create -f ` at any time to test it out on cluster with custom changes without deploying full stack.